cli/internal/cmd/apply.go

```diff
@@ -337,6 +337,18 @@ The control flow is as follows:
 	                        └────────────────────┘
 */
 func (a *applyCmd) apply(cmd *cobra.Command, configFetcher attestationconfigapi.Fetcher, upgradeDir string) error {
+	// Migrate state file
+	stateFile, err := state.ReadFromFile(a.fileHandler, constants.StateFilename)
+	if err != nil {
+		return fmt.Errorf("reading state file: %w", err)
+	}
+	if err := stateFile.Migrate(); err != nil {
+		return fmt.Errorf("migrating state file: %w", err)
+	}
+	if err := stateFile.WriteToFile(a.fileHandler, constants.StateFilename); err != nil {
+		return fmt.Errorf("writing state file: %w", err)
+	}
```

Comment:
We always write the state file back, even when we didn't change anything. We could add some "changed" boolean to the `Migrate` function or do the reading and writing inside the migration. The second option could be confusing since one needs to read the statefile to call methods on it.

Reply:
I think the current flow (as you've implemented it) is clear. I'd suggest leaving it like that.